packages/world/test/KeysInTableModule.t.sol

```diff
@@ -22,64 +22,106 @@ contract KeysInTableModuleTest is Test {
   KeysInTableModule keysInTableModule = new KeysInTableModule(); // Modules can be deployed once and installed multiple times
 
   bytes16 namespace = ROOT_NAMESPACE;
-  bytes16 sourceFile = bytes16("source");
+  bytes16 name = bytes16("source");
+  bytes16 singletonName = bytes16("singleton");
+  bytes16 compositeName = bytes16("composite");
   bytes32 key1 = keccak256("test");
   bytes32[] keyTuple1;
   bytes32 key2 = keccak256("test2");
   bytes32[] keyTuple2;
 
-  Schema sourceTableSchema;
-  Schema sourceTableKeySchema;
-  bytes32 sourceTableId;
+  Schema tableSchema;
+  Schema tableKeySchema;
+  Schema singletonKeySchema;
+  Schema compositeKeySchema;
+  bytes32 tableId;
+  bytes32 singletonTableId;
+  bytes32 compositeTableId;
 
   uint256 val1 = 123;
   uint256 val2 = 42;
 
   function setUp() public {
-    sourceTableSchema = SchemaLib.encode(SchemaType.UINT256);
-    sourceTableKeySchema = SchemaLib.encode(SchemaType.BYTES32);
+    tableSchema = SchemaLib.encode(SchemaType.UINT256);
+    tableKeySchema = SchemaLib.encode(SchemaType.BYTES32);
+    compositeKeySchema = SchemaLib.encode(SchemaType.BYTES32, SchemaType.BYTES32, SchemaType.BYTES32);
+
+    SchemaType[] memory _schema = new SchemaType[](0);
+    singletonKeySchema = SchemaLib.encode(_schema);
+
     world = IBaseWorld(address(new World()));
     world.installRootModule(new CoreModule(), new bytes(0));
     keyTuple1 = new bytes32[](1);
     keyTuple1[0] = key1;
     keyTuple2 = new bytes32[](1);
     keyTuple2[0] = key2;
-    sourceTableId = ResourceSelector.from(namespace, sourceFile);
   }
 
   function _installKeysInTableModule() internal {
     // Register source table
-    sourceTableId = world.registerTable(namespace, sourceFile, sourceTableSchema, sourceTableKeySchema);
+    tableId = world.registerTable(namespace, name, tableSchema, tableKeySchema);
+    singletonTableId = world.registerTable(namespace, singletonName, tableSchema, singletonKeySchema);
+    compositeTableId = world.registerTable(namespace, compositeName, tableSchema, compositeKeySchema);
 
     // Install the index module
     // TODO: add support for installing this via installModule
     // -> requires `callFrom` for the module to be able to register a hook in the name of the original caller
     // !gasreport install keys in table module
-    world.installRootModule(keysInTableModule, abi.encode(sourceTableId));
+    world.installRootModule(keysInTableModule, abi.encode(tableId));
+    world.installRootModule(keysInTableModule, abi.encode(singletonTableId));
+    world.installRootModule(keysInTableModule, abi.encode(compositeTableId));
   }
 
-  function testInstall() public {
+  function testFailInstallSingleton() public {
```

Comment:
should we add a quick comment here about that this is currently failing and why but we should get back to it in the future?